unitTests/controllers/MLFacetQueryControllerTest.ts

```diff
@@ -25,20 +27,25 @@ export function MLFacetQueryControllerTest() {
 
       queryBuilder = new QueryBuilder();
       mLFacetQueryController = new MLFacetQueryController(facet);
+      buildRequest();
+    }
+
+    function buildRequest() {
       mLFacetQueryController.putFacetIntoQueryBuilder(queryBuilder);
-      facetsRequest = queryBuilder.build().facets;
+      facetsRequests = queryBuilder.build().facets;
```

Comment:
I think the tests become easier to follow when using functions to retrieve values rather than storing values in an outer scope. e.g.

```
function facetRequests() {
    return queryBuilder.build().facets;
}

function latestFacetRequest() {
    const requests = facetRequests();
    return requests[requests.length - 1];
}
```